pkg/geo/geomfn/distance.go

```diff
@@ -635,7 +635,7 @@ func findPointSideOfLinearRing(point geodist.Point, linearRing geodist.LinearRin
 	// See also: https://en.wikipedia.org/wiki/Nonzero-rule
 	windingNumber := 0
 	p := point.GeomPoint
-	for edgeIdx := 0; edgeIdx < linearRing.NumEdges(); edgeIdx++ {
+	for edgeIdx, numEdges := 0, linearRing.NumEdges(); edgeIdx < numEdges; edgeIdx++ {
```

Comment:
Assigning the result of `linearRing.NumEdges()` seems to have shaved another ~1.5s off the running time for the benchmark. New profile: https://share.polarsignals.com/d3189b2/

Reply:
gee whiz. i wonder if we can optimise away calling `Edge` as well, considering it makes a copy (use the direct pointer to the edge instead if you can. it won't malloc anything because it's already initialised within the array!). give that a shot if you have time!

Reply:
Do you mind elaborating on this? Are you thinking of adding a receiver function to `(g *geomGeodistLinearRing)` that returns a `[][]float64` instead of what `Edge` returns? https://github.com/cockroachdb/cockroach/blob/b3dcc72e24077a4342ca75feede42e55ba20804c/pkg/geo/geomfn/distance.go#L292-L298

Reply:
something like that yeah. to be more precise it'd be returning `geom.Coord{flatCoords[i*stride:(i+1)*stride]}`.
but come to think it, it may still have to instantiate a `geodist.Edge` object which could be expensive (urgh!). we may have to break abstraction to prevent making a copy. i think maybe worth pursuing caching for now, unless it's super clean and obvious what to do?

Reply:
Yeah, I was also concerned about having to break the abstraction since it'd be kinda hacky. Let's skip it for now.